### PR TITLE
chore(liveness): Update state machine flow for no lights

### PR DESCRIPTION
**Pull request by esauerbo**
#### Description of changes
- Update `recording` to handle no lights challenge. (Move straight to `success` state after `hasFaceMatchedInOval` condition is met)
- Conditionally renders photosensitivity warning

#### Description of how you validated changes
- Ran unit tests locally
- Ran e2e tests locally
- Manually smoke tested 

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx [hunk 1]
```diff
@@ -109,6 +112,9 @@ export const LivenessCameraModule = (
 
   const [state, send] = useLivenessActor();
 
+  const isFaceMovementChallenge =
+    useLivenessSelector(selectChallengeType) === 'FaceMovementChallenge';
```
Comment (thaddmt) [inline packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx:116]: Can we store this as a constant somewhere?

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts [hunk 2]
```diff
@@ -397,6 +409,80 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
         },
         onDone: 'uploading',
       },
+      faceMovementCheck: {
```
Comment (thaddmt) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts:412]: Would it be possible to keep the checks combined and instead just skip the `flashFreshnessColors` step with a conditional based on the name of the check?

For example the `checkMatch` step could conditionally choose to go to the freshness color steps based on the challenge name?

Alternatively we can add conditionals to the freshness color steps to just skip them if the challenge name doesnt match?

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx (hunk 1 above)
Comment (esauerbo) [reply packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx:116]: You mean store a boolean for whether or not we're using that challenge?

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/__tests__/LivenessCameraModule.test.tsx [hunk 3]
```diff
@@ -360,6 +360,52 @@ describe('LivenessCameraModule', () => {
     expect(matchIndicator).toHaveLength(0);
   });
 
+  it('should render photosensitivity warning with FaceMovementAndLightChallenge and isNotRecording = true', async () => {
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/__tests__/LivenessCameraModule.test.tsx:363]:
```suggestion
  it('should render photosensitivity warning when challenge is FaceMovementAndLightChallenge and isNotRecording is true', async () => {
```

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/__tests__/LivenessCameraModule.test.tsx [hunk 4]
```diff
@@ -360,6 +360,52 @@ describe('LivenessCameraModule', () => {
...
+      />
+    );
+
+    const photosensitivityWarning = screen.queryByText(
+      instructionDisplayText.photosensitivityWarningHeadingText
+    );
+    expect(photosensitivityWarning).toBeInTheDocument();
+  });
+
+  it('should not render photosensitivity warning with FaceMovementChallenge and isNotRecording = true', async () => {
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/__tests__/LivenessCameraModule.test.tsx:386]: please adjust wording similar to above

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts (hunk 2 above)
Comment (esauerbo) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts]: Yes, updated to branch after `checkMatch` based on challenge type.

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx (hunk 1 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx]: Oh sorry i just meant `FaceMovementChallenge`

Comment (esauerbo) [reply packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx]: Ok cool, we're already exporting a constant for that so updated to actually use it lol

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts [hunk 5]
```diff
@@ -373,6 +377,18 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
               1: { target: 'ovalMatching' },
             },
           },
+          handleChallenge: {
+            always: [
+              {
+                target: 'delayBeforeFlash',
+                cond: 'hasFaceMovementAndLightChallenge',
```
Comment (thaddmt) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts:384]: Nit: would this be more correct as isFaceMovementAndLightChallenge?

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts [hunk 6]
```diff
@@ -373,6 +377,18 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
...
+              },
+              {
+                target: 'success',
+                cond: 'hasFaceMovementChallenge',
```
Comment (thaddmt) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts:388]: Nit: would this be more correct as isFaceMovementChallenge?

**Review by thaddmt — APPROVED** [review]
LGTM

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts (hunk 5 above)
Comment (esauerbo) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts:384]: Yeah I can update that

**Review by reesscot — APPROVED** [review]

**Review by zchenwei — APPROVED** [review]

**Pull request merged**